R/reportCalibration.R

```diff
@@ -0,0 +1,334 @@
+#' @returns data frame where column names have been replaced
+#'
+#'@importFrom dplyr %>% all_of rename_with
+
+.replaceVarName <- function(df) {
+  for (var in c("hs", "bs")) {
+    if (var %in% colnames(df)) {
+      df <- df %>%
+        rename_with(~ paste0(.x, "r"), all_of(var))
+    }
+  }
+  return(df)
+}
+
+#' Compute the deviation to historical data
+#'
+#' @param df data frame, containing data from calibration run
+#' @param dfHist data frame, containing historical data
+#' @returns data frame where the value column contains the deviation from historical data
+#'
+#' @importFrom dplyr %>% .data filter full_join mutate rename select
+#' @importFrom tidyr crossing replace_na
+
+.computeDeviation <- function(df, dfHist) {
+
+  # Add iteration column to historical data and filter for time periods of calibration data in df
+  dfHist <- dfHist %>%
+    crossing(iteration = df[["iteration"]]) %>%
+    filter(.data[["ttot"]] %in% unique(df[["ttot"]])) %>%
+    rename(hist = "value")
+
+  # Replace all data points not present in either data frames by zero and compute the difference
+  df <- df %>%
+    full_join(dfHist,
+              by = intersect(colnames(df), colnames(dfHist))) %>% # Rather unsure about this ...
+    replace_na(list(value = 0, hist = 0)) %>%
+    mutate(value = .data[["value"]] - .data[["hist"]]) %>%
+    select(-"hist")
+
+}
+
+#' Compute the sum of the squares in a data frame
+#'
+#' @param df data frame, containing the data to be evaluated
+#' @param rprt character, column names for which the sum of the squares should be reported separately
+#' @returns data frame sum of squares as value column
+#'
+#' @importFrom dplyr %>% across all_of any_of .data group_by rename_with summarise ungroup
+
+.computeSumSq <- function(df, rprt = "") {
+
+  df %>%
+    group_by(across(any_of(rprt))) %>%
+    summarise(value = sum(.data[["value"]] ^ 2, na.rm = TRUE), .groups = "drop")
+
+}
+
+#' Compute the relative deviation to historical data as the relative euclidean distance
+#'
+#' @param dfDev data frame, containing deviation between calibration and historical data
+#' @param dfHist data frame or a list of one or two data frames, containing historical data.
+#'  The option to pass two data frames is used to combine construction and renovation flows.
+#'  If two data frames are passed, the first one is assumed to be construction, the second renovation
+#' @param tCalib numerical/factor, calibration time periods to filter historical data
+#' @param notInHist character vector, pass columns that historical data should not be grouped by
+#'  when computing the sum of the squares
+#' @returns data frame with relative deviation in value column
+#'
+#' @importFrom dplyr %>% .data filter left_join mutate rename select
+
+.computeRelDev <- function(dfDev, dfHist, tCalib, notInHist = NULL) {
+
+  # Determine the reported columns in the calibration data
+  rprt <- setdiff(colnames(dfDev), c("iteration", "value", notInHist))
```

Comment:
This function was adapted to allow for the modified computation of relative deviations by hs.

Reply:
looking good